src/entt/entity/storage.hpp

```diff
@@ -233,25 +307,41 @@ template<typename... Lhs, typename... Rhs>
  * @tparam Allocator Type of allocator used to manage memory and elements.
  */
 template<typename Type, typename Entity, typename Allocator, typename>
-class basic_storage: public basic_sparse_set<Entity, typename std::allocator_traits<Allocator>::template rebind_alloc<Entity>> {
+class basic_storage
+    : public basic_sparse_set<Entity, typename std::allocator_traits<Allocator>::template rebind_alloc<Entity>>,
+      protected Allocator,
+      protected component_traits<Type> {
+    template<typename Container, std::size_t Size>
+    friend struct internal::static_storage_accessor;
+    template<typename Storage>
+    friend struct internal::dynamic_storage_accessor;
```

Comment:
These `friend`s are a code smell that makes me wonder again if these _accessors_ aren't superfluous.